### PR TITLE
refactor: replaced moment with dayjs

### LINKED ISSUE #125: Swap moment with dayjs
`moment` accounts for over 60% of this packages size.

[moment](https://github.com/moment/moment/): 65.9KB
[dayjs](https://github.com/iamkun/dayjs): 2KB

### PR DESCRIPTION
**Pull request by GoPro16**
BREAKING CHANGE: moment has been replaced

All of the existing tests that cover the date validation work. The only downside is that we will have to release as a `BREAKING CHANGE`  since some users can rely upon `moment` in other areas of their projects since its a required dep here.

closes #125

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
   "dependencies": {
     "babel-runtime": "^6.26.0",
     "classnames": "^2.2.6",
+    "dayjs": "^1.8.11",
     "lodash": "^4.17.10",
     "moment": "^2.22.2",
```
Comment (TheSharpieOne) [inline package.json]: Probably want to remove moment as a dep.

**Comment by TheSharpieOne** [comment]
I don't think this is a breaking change. We don't expose the underlying date library and if someone is depending on something without it being in their owns deps... then that is their fault.

**Comment by GoPro16** [comment]
For some reason `dayjs` customParseFormat is able to parse an invalid parse format to a valid date.
https://github.com/Availity/availity-reactstrap-validation/blob/897a4af446b4960d98a5c917c3df5023d5e5547a/__test__/AvValidator.date.spec.js#L19

Passing in `10/12/2014` with parse format of `YYYY-MM-DD` results in the valid date of `08-14-2015`

Not sure why this is happening but I am going to open an issue on `dayjs` for this. This and one other test that is similar failed.

**Comment by GoPro16** [comment]
Created issue here: https://github.com/iamkun/dayjs/issues/555

**Comment by robmcguinness** [comment]
Fixed in https://github.com/iamkun/dayjs/releases/tag/v1.8.13

**Pull request merged**